packages/mock-block-dock/src/datastore/mutate-subgraph.ts

```diff
@@ -0,0 +1,123 @@
+import {
+  Entity,
+  EntityId,
+  EntityVertex,
+  KnowledgeGraphOutwardEdge,
+  KnowledgeGraphRootedEdges,
+  KnowledgeGraphVertices,
+  Subgraph,
+  Timestamp,
+} from "@blockprotocol/graph";
+import isEqual from "lodash/isEqual";
+
+/** @todo - clean up the assertions here */
```

Comment:
I'm actually not sure why there are so many.

I think a lot of code like this could be fixed up with a `addOrSetWithDefault` utility method or something